sycl/include/CL/sycl/intel/reduction.hpp

```diff
@@ -408,14 +470,30 @@ class reduction_impl {
                     access::target::global_buffer>(*RWReduBuf, CGH);
   }
 
-  /// User's accessor to where the reduction must be written.
-  accessor_type MAcc;
+  accessor_type &getUserAccessor() { return MAcc; }
+
+  T *getUSMPointer() {
+    assert(is_usm && "Unexpected call of getUSMPointer().");
+    return MUSMPointer;
+  }
```

Comment:
This is a question, not a request for a change.  Would it make sense to use SFINAE to disable these in cases that they won't work, or do you think this is not worth the pain (because users will never call these functions anyway)?

Reply:
I thought about that, but decided not adding SFINAE for this method now because

a) reduction_impl lives on HOST and it is not such a big overhead for it to carry MUSMPointer field and have this method also. Also, user is not supposed to call this method, it is closed.

b) adding SFINAE to getUSMPointer will require to have split parallel_for() at handler.hpp:854 because I wouldn't be able to have that simple if handler.hpp:863.
Similar for another parallel_for using that method (handler.hpp:983)

So, I thought it is more code without clear benefit.

Reply:
Ok, that makes sense.  Thanks for the explanation.